### PR TITLE
Bugfix/jr2 1171 location postalcode notblank

### PR DESCRIPTION
**Pull request by cjpsycos**

### CODE REVIEW

File: domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/Company.java [hunk 1]
```diff
@@ -36,9 +36,6 @@ private Company(Builder builder) {
         this.houseNumber = builder.houseNumber;
         this.postalCode = builder.postalCode;
         this.city = builder.city;
-        // FIXME Add constrains when legacy API is removed!
-//        this.postalCode = Condition.notBlank(builder.postalCode, "Postal code of a company can't be null");
-//        this.city = Condition.notBlank(builder.city, "City of a company can't be null");
```
Comment (cjpsycos) [inline domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/Company.java]: from AVAM it comes postalcode+city or poboxPostalcode+poboxCity

File: application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/JobAdvertisementApplicationService.java [hunk 2]
```diff
@@ -467,23 +465,18 @@ private JobAdvertisementCreator getJobAdvertisementCreatorFromInternal(CreateJob
         Location location = toLocation(createJobAdvertisementDto.getLocation());
         location = locationService.enrichCodes(location);
 
-        String jobCenterCode = jobCenterService.findJobCenterCode(location.getCountryIsoCode(), location.getPostalCode());
+        Condition.notNull(createJobAdvertisementDto.getOccupation(), "Occupation can't be null");
```
Comment (birom) [inline application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/JobAdvertisementApplicationService.java]: There is a todo in the CreateJobAdvertisementDto#38. We need to add the @NotNull constraint to the  CreateJobAdvertisementDto.occupation property.

Comment (cjpsycos) [reply application/src/main/java/ch/admin/seco/jobs/services/jobadservice/application/jobadvertisement/JobAdvertisementApplicationService.java]: That's already done

File: domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/JobContent.java [hunk 3]
```diff
@@ -131,8 +131,7 @@ public JobContent(Builder builder) {
         this.publicContact = builder.publicContact;
         this.applyChannel = builder.applyChannel;
         this.location = builder.location;
-        // FIXME reinsert occupation condition, when the legacy API is deleted
-        //this.occupations = Condition.notEmpty(builder.occupations, "Occupations can't be null or empty");
+        this.occupations = Condition.notEmpty(builder.occupations, "Occupations can't be null or empty");
```
Comment (cjpsycos) [inline domain/src/main/java/ch/admin/seco/jobs/services/jobadservice/domain/jobadvertisement/JobContent.java]: Remove Condition because of X28

**Pull request merged**